mpxe/harness/project.py

```diff
@@ -0,0 +1,68 @@
+import subprocess
+import os
+import fcntl
+import signal
+import importlib
+import sys
+
+from mpxe.harness import decoder
+from mpxe.protogen import stores_pb2
+from mpxe.sims import sim
+
+class Project:
+    def __init__(self, name, sim):
+        self.name = name
+        self.killed = False
+        self.stores = {}
+
+        cmd = 'build/bin/x86/{}'.format(self.name)
+        self.popen = subprocess.Popen(cmd, bufsize=0, shell=False, stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=False)
+        ctop_fifo_path = '/tmp/{}_ctop'.format(self.popen.pid)
+        while not os.path.exists(ctop_fifo_path):
+            pass
+
+        # Subprocess is expected to create child to parent fifo, open in raw bytes mode
+        self.ctop_fifo = open(ctop_fifo_path, 'rb')
+        if self.ctop_fifo is None:
+            raise Exception('failed to open ctop fifo')
+        ctop_fd = self.ctop_fifo.fileno()
+        ctop_fl = fcntl.fcntl(ctop_fd, fcntl.F_GETFL)
+        fcntl.fcntl(ctop_fd, fcntl.F_SETFL, ctop_fl | os.O_NONBLOCK)
+
+        self.sim = sim
+        print('started', self.name)
+
+    def stop(self):
+        if self.killed:
+            return
+        self.popen.terminate()
+        self.popen.wait()
+        self.popen.stdout.close()
+        self.popen.stdin.close()
+        self.ctop_fifo.close()
+        ctop_fifo_path = '/tmp/{}_ctop'.format(self.popen.pid)
+        os.unlink(ctop_fifo_path)
+        self.killed = True
+        print('stopped', self.name)
+
+    def write_store(self, store_type, msg, mask):
+        update = stores_pb2.MxStoreUpdate()
+        update.key = 0
```

Comment:
should `write_store` take an optional `key` keyword arg to support writing with keys?